src/pgwire/src/protocol.rs

```diff
@@ -876,13 +876,20 @@ where
         .await?;
         ROWS_RETURNED.inc_by(u64::cast_from(nrows));
 
-        if rows.is_empty() {
+        // Always return rows back, even if it's empty. This prevents an unclosed
+        // portal from re-executing after it has been emptied.
+        portal.set_remaining_rows(rows);
+
+        // If max_rows is not specified, we will always send back a CommandComplete. If
+        // max_rows is specified, we only send CommandComplete if there were more rows
+        // requested than were remaining. That is, if max_rows == num remaining rows,
```

Comment:
im a little bit confused by this comment / the surrounding code because it makes it seem like 

"rows requested" matches the variable `max_rows`
and
"[number of rows] remaining" matches `nrows`

based on the the clause "we only send CommandComplete ..."


but it seems to me that the number of rows remaining is now `rows.len()` and `nrows` is the count of rows that have already been sent (and based on line 870 it seems like `max_rows > nrows` can never evaluate to true?

I haven't read the full file so I could easily be missing something - let me know!

Reply:
This was due to confusion in the comment around terminology about the word remaining. The comment always refers to rows remaining before sending, not remaining rows after sending. I've added some text to the comment to try to clarify that, but it may have made it more confusing. How does it read now?

Reply:
got it the new comment clears that up